README.md

```diff
@@ -536,6 +541,166 @@ webSocket.done.then((_) {
 
 ---
 
+## Testing & Mocks
+
+Just like the browser or the Dart VM, tests are considered a platform for which
+this library can be configured. By configuring `w_transport` for tests, mock
+implementations of all classes will be used.
+
+```dart
+import 'package:w_transport/w_transport_mock.dart';
+
+main() {
+  configureWTransportForTest();
+}
+```
+
+That's it. No changes to your source code are necessary! Once configured for
+test, you are in control of every HTTP request and every WebSocket connection.
+The API for controlling these transports are exported with the
+`w_transport_mock.dart` entry point as static APIs on a `MockTransports` class.
+
+> **Resetting Mocks:**
+>
+> At any point, you can reset all mock expectations and handlers, giving you a
+> clean state to begin a new mock setup:
+>
+> ```dart
+> MockTransports.reset();
+> ```
+
+### Mocking HTTP
+
+#### Expecting a request (and returning a 200 OK response by default)
+```dart
+MockTransports.http.expect('GET', Uri.parse('/resource'));
+```
+
+#### Expecting a request and providing a custom response
+```dart
+Response response = new MockResponse.unauthorized(
+    body: 'Invalid access token',
+    headers: {'x-session': 'ab93s...'});
+MockTransports.http.expect(
+    'GET',
+    Uri.parse('/resource'),
+    respondWith: response);
+```
+
+#### Expecting a request and causing a failure (exception)
+```dart
+MockTransports.http.expect(
+    'GET',
+    Uri.parse('/resource'),
+    failWith: new Exception('Unexpected error...'));
+```
+
+#### Registering a handler (expecting multiple requests to the same URI)
+```dart
+MockTransports.http.when(
+    Uri.parse('/resource'),
+    (FinalizedRequest request) async {
+      if (request.method == 'GET') {
+        return new MockResponse.ok(body: '...');
+      }
+      if (request.method == 'DELETE') {
+        return new MockResponse(204);
+      }
+      ...
+    });
+```
+
+#### Registering a handler (expecting multiple request with the same URI and method)
+```dart
+MockTransports.http.when(
+    Uri.parse('/resource'),
+    (FinalizedRequest request) async => new MockResponse.ok(body: '...'),
+    method: 'GET');
+```
+
+#### Verifying there are no unresolved requests and/or unsatisfied expectations
+```dart
+MockTransports.verifyNoOutstandingExceptions();
```

Comment:
I borrowed this name from angulars mocks from way back.. the reason it uses the term `Exceptions` instead of `Expectations` is because there are two things it looks for:

1. Unresolved requests - requests that where sent that were not expected
2. Unsatisfied expectations - requests that never happened but were supposed to

Only one of those is really an expectation, the other is unexpected. That being said, exceptions isn't the most descriptive name either..